modules/core/include/opencv2/core/dualquaternion.hpp

```diff
@@ -487,6 +524,77 @@ class DualQuat{
+     * \f$w = (w_1,...,w_n)\f$, the generalized DQB is simply
+     * \f[
+     * gDQB(\boldsymbol{w};{\boldsymbol{q}}_1,...,{\boldsymbol{q}}_n)=\frac{w_1{\boldsymbol{q}}_1+...+w_n{\boldsymbol{q}}_n}
+     * {||w_1{\boldsymbol{q}}_1+...+w_n{\boldsymbol{q}}_n||}.
+     * \f]
+     * @param dualquat vector of dual quaternions
+     * @param weights vector of weights. \f$\sum_0^n w_{i} = 1\f$ and \f$w_i>0\f$
+     * @param assumeUnit if @ref QUAT_ASSUME_UNIT, these dual quaternions assume to be unit quaternions
+     * and this function will save some computations.
+     */
+    static DualQuat<_Tp> gdqblend(const std::vector<DualQuat<_Tp>> &dualquat, const std::vector<_Tp> &weights,
+                                QuatAssumeType assumeUnit=QUAT_ASSUME_NOT_UNIT);
+
+    /**
+     * This function is a skinning algorithm to deform the mesh with dual quaternion skinning(DQS) deformer with reference
+     * to the paper ["Geometric Skinning with Approximate Dual Quaternion Blending"](https://www.cs.utah.edu/~ladislav/kavan08geometric/kavan08geometric).
+     * We use the generalized Dual Quaternion linear Blending to compute the deformed position:
+     * \f[
+     * DQB(\boldsymbol{w};{\boldsymbol{q}}_1,...,{\boldsymbol{q}}_n)=
+     * \frac{w_1{\boldsymbol{q}}_1+...+w_n{\boldsymbol{q}}_n}{||w_1{\boldsymbol{q}}_1+...+w_n{\boldsymbol{q}}_n||}.
+     * \f]
+     * And it will always choose the shortest rotation path. Compared with Linear Blending Skinning(LBS),
+     * the DQS can avoid the loss of volume and have a similar performance on run-time.
+     *
+     * @param in_vert vector of vertices at original position
+     * @param in_normals vector of mesh normals
+     * @param out_vert deformed vertices transformed by dual quaternions
+     * @param out_normals deformed mesh normals transformed by dual quaternions
+     * @param dualquat vector of unit dual quaternions for each joint
+     * @param weights vector of influence weights for each vertex. All weights that influence one vertex should
+     * satisfy \f$\sum_0^n w_{i} = 1\f$ and \f$w_i>0\f$
+     * @param joints_id vector of joints id that influence one vertex for each vertex(same order as weights).
+     * The ID here represents the order of dual quaternions. So the joints id should be non-negative.
+     * @param assumeUnit if @ref QUAT_ASSUME_UNIT, these dual quaternions assume to be unit quaternions
+     * and this function will save some computations.
+     *
+     * @sa gdqblend
+     *
+     */
+    static void dqs(const std::vector<Vec<_Tp, 3>> &in_vert,
+                  const std::vector<Vec<_Tp, 3>> &in_normals,
+                  std::vector<Vec<_Tp, 3>> &out_vert,
+                  std::vector<Vec<_Tp, 3>> &out_normals,
+                  const std::vector<DualQuat<_Tp>> &dualquat,
+                  const std::vector<std::vector<_Tp>> &weights,
+                  const std::vector<std::vector<int>> &joints_id,
```

Comment:
It's better to use `InputArray` and `OutputArray` for vertices, normals, weights and ids instead of vectors. Internally the type and dimensionality should be checked.

Reply:
OK